dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -324,6 +333,92 @@ default String getURLFromEntry(String dockerEntry, ToolDAO toolDAO) {
         return url;
     }
 
+    // TODO: Implement for DockerHub, then gitlab and seven bridges;
+    default Set<Image> getImagesFromRegistry(String toolsJSONTable) {
+        ArrayList<Map<String, String>> dockerTools = new ArrayList<>();
+        Gson gson = new Gson();
+        dockerTools = (ArrayList<Map<String, String>>)gson.fromJson(toolsJSONTable, dockerTools.getClass());
+
+        Set<String> dockerStrings = new HashSet<>();
+
+        // Eliminate duplicate docker strings
+        for (Map<String, String> tool : dockerTools) {
+            dockerStrings.add(tool.get("docker"));
+        }
+
+        Set<Image> dockerImages = new HashSet<>();
+        for (String image : dockerStrings) {
+            String[] parts = image.split("/");
+
+            Optional<String> response;
+            if (image.startsWith("quay.io/")) {
+                String errorKey = "error_message";
+                String[] splitDocker = image.split("/");
+                String[] splitTag = splitDocker[2].split(":");
+                if (splitTag.length > 1) {
+                    String repo = splitDocker[1] + "/" + splitTag[0];
+                    String tagName = splitTag[1];
+                    response = getImageResponseFromQuay(image);
+
+                    if (response.isPresent()) {
+                        Map<String, ArrayList<Map<String, String>>> map = new HashMap<>();
+                        Map<String, String> errorMap = new HashMap<>();
+                        map = (Map<String, ArrayList<Map<String, String>>>)gson.fromJson(response.get(), map.getClass());
+                        errorMap = (Map<String, String>)gson.fromJson(response.get(), errorMap.getClass());
+                        if (errorMap.get(errorKey) != null) {
+                            LOG.error("Error response from Quay: " + errorMap.get(errorKey));
+                        } else {
+                            try {
+                                final List<Map<String, String>> array = map.get("tags");
+
+                                for (Map<String, String> tag : array) {
+                                    final String digest = tag.get("manifest_digest");
+                                    final String imageID = tag.get("image_id");
+                                    List<Checksum> checksums = new ArrayList<>();
+                                    checksums.add(new Checksum(digest.split(":")[0], digest.split(":")[1]));
+                                    dockerImages.add(new Image(checksums, repo, tagName, imageID));
+                                }
+
+                            } catch (IndexOutOfBoundsException | NullPointerException ex) {
+                                LOG.error("Could not get checksum information for " + splitDocker[1]);
```

Comment:
Thought, this could be hooked up to a validation warning when we have it. 
Further food for thought in https://github.com/dockstore/dockstore/issues/3216